### PR TITLE
[tools revamp] Add some P0 features

### PR DESCRIPTION
**Pull request by chejennifer**
- allow specifying denom in the url for map and scatter variables
- allow specifiying date in the url for map
- add the get BQ SQL Query button for map
- add zoom functionality for map

screen-recording-2023-08-04T084540 036

### CODE REVIEW

File: static/js/tools/map/bq_query_utils.ts [hunk 1]
```diff
@@ -180,8 +183,10 @@ function getBaseWhereClause(
 }
 
 export function getPcQuery(
```
Comment (shifucun) [inline static/js/tools/map/bq_query_utils.ts:185]: looks like the BQ modal is in many tools not only map. Should these live outside the map folder?

**Review by shifucun — APPROVED** [review]
map zoom is a very nice feature! A highly asked one from friends testing :)

File: static/js/tools/map/bq_query_utils.ts (hunk 1 above)
Comment (chejennifer) [reply static/js/tools/map/bq_query_utils.ts:185]: The BQ modal lives in the shared folder. These are the query utils specific to getting the query for the map tool!

**Comment by chejennifer** [comment]
thanks for the review!

**Pull request merged**